pkg/cache/cache.go

```diff
@@ -0,0 +1,26 @@
+// Copyright © 2020 by PACE Telematics GmbH. All rights reserved.
+// Created at 2020/08/12 by Marius Neugebauer
+
+package cache
+
+import (
+	"context"
+	"time"
+)
+
+// Strings caches strings.
+type Strings interface {
```

Comment:
I would change the interface to operate on bytes, like most other kv stores do:

* https://godoc.org/github.com/cznic/kv
* https://pkg.go.dev/bazil.org/bazil/kv?tab=doc
* https://pkg.go.dev/github.com/syndtr/goleveldb/leveldb?tab=doc
Concrete: 

```go
Get(ctx context.Context, key []byte) ([]byte, error)
Put(ctx context.Context, key, value []byte) error
PutTTL(ctx context.Context, key, value []byte, ttl time.Duration) error
```

Reply:
We are free to add another interface, that works on bytes, if needed. But this one works on strings. I see this package as a convenience layer on top of other key-value stores. There wouldn't be much convenience if I still need to cast everything.

Reply:
I added some package documentation about the purpose of this package.

Reply:
Ok, the following problem exists with my idea. If for example the Redis implementation should support both strings and ints, then it anyway needs some form of abstraction to distinguish strings from strings that encode ints. And one would be able to request a string as an int, which could fail. But this error category would be unclear to the caller. Also one could store a string containing a number and request it as int. So it works even though it shouldn't. I don't like this error category, so I will think about the bytes approach once more.

Reply:
I'll change the interface to use `string` for the key and `[]byte` for the value.

Reply:
Done.